Metalama.Framework.Engine/Templating/TemplatingCodeValidator.Visitor.cs

```diff
@@ -572,6 +574,17 @@ public void Report( Diagnostic diagnostic )
                     this.HasError = true;
                 }
 
+                if ( this._isDesignTime && diagnostic.Location.IsInSource && !this._semanticModel.Compilation.ContainsSyntaxTree( diagnostic.Location.SourceTree ) )
+                {
+                    // At design-time, we have to report diagnostics on the correct tree instance.
+                    // (At compile-time, Metalama Compiler ignores which tree object a diagnostic is reported on.)
+
+                    var targetTree = this._semanticModel.SyntaxTree;
+
+                    // It is a hack to use manifest code here, but it already does what we need.
+                    diagnostic = new CompileTimeDiagnosticManifest( diagnostic, new() { { targetTree.FilePath, 0 } } ).ToDiagnostic( [targetTree] );
```

Comment:
The mapping logic should be in the caller and not here (it's not a concern of the validator but of the diagnostic sinc). I.e. the logic must be in the in the _reportDiagnostic delegate passed by the caller.